### PR TITLE
Documentation Dump from a side project

### PR DESCRIPTION
**Pull request by adrienntindall**

### CODE REVIEW

File: src/comm_ring.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    ring->buffer = buffer;
+    ring->size = size;
+    ring->startIndex = 0;
+    ring->endIndex = 0;
+    ring->backupEndIndex = 0;
+}
+
+void CommRring_Write (CommRing * ring, u8 * buffer, int size, int unused)
+{
+    int i, j;
```
Comment (lhearachel) [inline src/comm_ring.c:21]: Can these by declared in-line C99 style?

File: src/comm_ring.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        GF_ASSERT(buffer);
+        ring->buffer[CommRing_Index(ring, i)] = buffer[j];
+    }
+
+    ring->backupEndIndex = CommRing_Index(ring, i);
+}
+
+int CommRing_Read (CommRing * ring, u8 * buffer, int size)
+{
+    int index;
```
Comment (lhearachel) [inline src/comm_ring.c:40]: Ditto

File: src/comm_ring.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    u8 buff;
+
+    CommRing_Read(ring, &buff, 1);
+    return buff;
+}
+
+//Reading but doesn't incriment the index
+int CommRing_Peek (CommRing * ring, u8 * buffer, int size)
+{
+    int i, j;
```
Comment (lhearachel) [inline src/comm_ring.c:59]: Ditto

File: src/overlay005/fieldmap.c [hunk 4]
```diff
@@ -0,0 +1,1002 @@
...
+    FieldSystem * fieldSystem = param0;
+
+    sub_0201C2B8(fieldSystem->unk_08);
+    sub_0201DCAC();
+    sub_0200A858();
+
+    inline_ov5_021D0D80(fieldSystem);
+}
+
+static int FieldMap_Init (OverlayManager * param0, int * param1)
```
Comment (lhearachel) [inline src/overlay005/fieldmap.c:154]: The `Init`, `Main`, and `Exit` functions should return `BOOL` to align with the `OverlayFunc` typedef.

File: src/overlay005/fieldmap.c [hunk 5]
```diff
@@ -0,0 +1,1002 @@
...
+    fieldSystem = OverlayManager_Args(param0);
+
+    if (ov5_021D119C(fieldSystem)) {
+        sub_02055D94(fieldSystem);
+        ov5_021D13B4(fieldSystem);
+        ov5_021EA714(fieldSystem, 1, 1);
+
+        if (FieldMap_ChangeZone(fieldSystem)) {
+            (void)0;
+        }
```
Comment (lhearachel) [inline src/overlay005/fieldmap.c:263]: Does the compiler optimize this out?

**Review by lhearachel — COMMENTED** [review]
Just a brief first pass

File: src/comm_ring.c (hunk 1 above)
Comment (adrienntindall) [reply src/comm_ring.c:21]: this function specifically i gets used out of the for loop so no

File: src/player_avatar.c [hunk 6]
```diff
@@ -0,0 +1,922 @@
...
+}
+
+void PlayerAvatar_SetInDeepSwamp (PlayerAvatar * param0, int param1)
+{
+    if (param1 == 1) {
+        sub_0205EB9C(param0, (1 << 4));
+    } else {
+        sub_0205EBA4(param0, (1 << 4));
+    }
+}
```
Comment (lhearachel) [inline src/player_avatar.c:787]: `BOOL flag` for `param1`? (and update invocations to use `TRUE`/`FALSE`)

File: src/player_avatar.c [hunk 7]
```diff
@@ -0,0 +1,922 @@
...
+
+int PlayerAvatar_IsNotInDeepSwamp (PlayerAvatar * param0)
+{
+    if (sub_0205EBB0(param0, (1 << 4))) {
+        return 1;
+    }
+
+    return 0;
+}
```
Comment (lhearachel) [inline src/player_avatar.c:796]: Make this return `BOOL`

File: src/player_avatar.c [hunk 8]
```diff
@@ -0,0 +1,922 @@
...
+
+int sub_0205F034 (PlayerAvatar * param0)
+{
+    if (sub_0205EBB0(param0, (1 << 5))) {
+        return 1;
+    }
+
+    return 0;
+}
```
Comment (lhearachel) [inline src/player_avatar.c:814]: Should return `BOOL`

File: src/player_avatar.c [hunk 9]
```diff
@@ -0,0 +1,922 @@
...
+
+int sub_0205F060 (PlayerAvatar * param0)
+{
+    if (sub_0205EBB0(param0, (1 << 6))) {
+        return 1;
+    }
+
+    return 0;
+}
```
Comment (lhearachel) [inline src/player_avatar.c:833]: Should return `BOOL`

File: src/unk_0205F180.c [hunk 10]
```diff
@@ -1161,44 +1159,44 @@ static void sub_02060258 (PlayerAvatar * param0, LocalMapObject * param1, int pa
...
+static void PlayerAvatar_TryCyclingGearChange (PlayerAvatar * playerAvatar, u16 pad)
 {
-    if (sub_0205EB74(param0) != 0x1) {
+    if (PlayerAvatar_PlayerState(playerAvatar) != 0x1) {
         return;
     }
 
-    if ((param1 & PAD_BUTTON_B)) {
-        u32 v0 = sub_02062BE8(Player_LocalMapObject(param0));
+    if ((pad & PAD_BUTTON_B)) {
```
Comment (lhearachel) [inline src/unk_0205F180.c:1183]: Extraneous parens

File: src/unk_0205F180.c [hunk 11]
```diff
@@ -978,66 +976,66 @@ static void sub_0205FDC8 (PlayerAvatar * param0, LocalMapObject * param1, int pa
...
-            UnkStruct_0205EC34 * v7;
+            PlayerData * v7;
 
             v1 = 0xc;
             v2 = 4;
-            v7 = sub_0205EC0C(param0);
+            v7 = PlayerAvatar_PlayerData(playerAvatar);
 
-            if ((sub_0205EC40(v7) == 1) && (sub_020613FC(param0, param4) == 1)) {
+            if ((PlayerData_HasRunningShoes(v7) == 1) && (sub_020613FC(playerAvatar, param4) == 1)) {
```
Comment (lhearachel) [inline src/unk_0205F180.c:1025]: Extraneous parens; replace `== 1` with `== TRUE`

File: src/unk_0205F180.c [hunk 12]
```diff
@@ -484,190 +482,190 @@ static const UnkStruct_020EDB84 Unk_020EDB84[] = {
...
-    LocalMapObject * v1 = Player_LocalMapObject(param0);
-    int v2 = sub_0206299C(v1);
+    MapObject * v1 = Player_MapObject(playerAvatar);
+    int v2 = MapObject_MoveDir(v1);
 
     Sound_PlayEffect(1620);
 
     if (v2 == 0) {
-        if ((sub_0205EB74(param0) == 0x1) && (sub_0205EBB8(param0) >= 3)) {
+        if ((PlayerAvatar_PlayerState(playerAvatar) == 0x1) && (PlayerAvatar_Speed(playerAvatar) >= 3)) {
```
Comment (lhearachel) [inline src/unk_0205F180.c:568]: Extraneous parens

File: src/unk_0205F180.c [hunk 13]
```diff
@@ -2456,36 +2454,36 @@ void sub_02061674 (PlayerAvatar * param0, int param1, int * param2, int * param3
...
     } else {
-        LocalMapObject * v1 = Player_LocalMapObject(param0);
-        FieldSystem * v2 = sub_02062C00(v1);
-        int v3 = sub_02063020(v1);
-        int v4 = ((sub_02063030(v1) / 2));
-        int v5 = sub_02063040(v1);
+        MapObject * v1 = Player_MapObject(playerAvatar);
+        FieldSystem * v2 = MapObject_FieldSystem(v1);
+        int v3 = MapObject_XPos(v1);
+        int v4 = ((MapObject_YPos(v1) / 2));
```
Comment (lhearachel) [inline src/unk_0205F180.c:2467]: Extraneous parens

File: src/unk_0205F180.c [hunk 14]
```diff
@@ -2456,36 +2454,36 @@ void sub_02061674 (PlayerAvatar * param0, int param1, int * param2, int * param3
...
     u32 v0;
-    LocalMapObject * v1 = Player_LocalMapObject(param0);
-    FieldSystem * v2 = sub_02062C00(v1);
-    int v3 = sub_02063020(v1);
-    int v4 = ((sub_02063030(v1) / 2));
-    int v5 = sub_02063040(v1);
+    MapObject * v1 = Player_MapObject(playerAvatar);
+    FieldSystem * v2 = MapObject_FieldSystem(v1);
+    int v3 = MapObject_XPos(v1);
+    int v4 = ((MapObject_YPos(v1) / 2));
```
Comment (lhearachel) [inline src/unk_0205F180.c:2483]: Extraneous parens

**Review by lhearachel — COMMENTED** [review]
Limiting my feedback to files that you've very obviously touched yourself and not those that got hit by mass renames

File: src/overlay005/ov5_021DFB54.c [hunk 15]
```diff
@@ -160,334 +160,332 @@ static void ov5_021E0FF0(SysTask * param0, void * param1);
...
 static void ov5_021E1134(void * param0);
 static Pokemon * ov5_021E1140(FieldSystem * param0, int param1);
 static void ov5_021E0DE0(FieldSystem * param0);
 static BOOL ov5_021E0E10(TaskManager * param0);
 
-static void(*const Unk_ov5_021F9B74[10])(PlayerAvatar *);
+static void(*const sPlayerAvatarRequestStateTbl[10])(PlayerAvatar *);
 int(*const Unk_ov5_021F9B54[])(UnkStruct_ov5_021F9B54 *);
 int(*const Unk_ov5_021F9B10[])(UnkStruct_ov5_021F9B10 *);
 int(*const Unk_ov5_021F9AFC[])(UnkStruct_ov5_021F9B10 *);
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/overlay005/ov5_021DFB54.c:175]: It's not super important and the typedef would only be used here

File: src/overlay005/ov5_021EB1A0.c [hunk 16]
```diff
@@ -69,39 +69,39 @@ typedef struct {
...
+static void ov5_021EC374(MapObject * param0, UnkStruct_020216E0 * param1, int param2);
 
-void(*const Unk_ov5_021FF3D4[])(LocalMapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EB2EC *, int);
-void(*const Unk_ov5_021FF420[])(LocalMapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EB2EC *, int);
-void(*const Unk_ov5_021FF3F8[])(LocalMapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EBA0C *, int);
-static void(*const Unk_ov5_021FF448[11])(LocalMapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EBA0C *, int);
+void(*const Unk_ov5_021FF3D4[])(MapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EB2EC *, int);
+void(*const Unk_ov5_021FF420[])(MapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EB2EC *, int);
+void(*const Unk_ov5_021FF3F8[])(MapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EBA0C *, int);
+static void(*const Unk_ov5_021FF448[11])(MapObject *, UnkStruct_020216E0 *, UnkStruct_ov5_021EBA0C *, int);
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/overlay005/ov5_021EB1A0.c:102]: Maybe but this isn't a file I've touched beyond auto replace (and now I just changed all MapObject * param0's -> mapObk)

File: src/player_avatar.c [hunk 17]
```diff
@@ -0,0 +1,922 @@
...
+{
+    MapObject * mapObj;
+
+    mapObj = Player_MapObject(playerAvatar);
+    GF_ASSERT(mapObj != NULL);
+    sub_02062A40(mapObj);
+
+    ov5_021F6218(playerAvatar);
+
+    if (PlayerAvatar_PlayerState(playerAvatar) == 0x2) {
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/player_avatar.c:103]: yup I just didn't add the const for 0x2 until after I moved on from this file, good catch

File: src/player_avatar.c [hunk 18]
```diff
@@ -0,0 +1,922 @@
...
+
+    PlayerAvatar_SetMapObject(playerAvatar, mapObj);
+}
+
+MapObject * sub_0205EA24 (const MapObjectManager * mapObjMan)
+{
+    int v0 = 0;
+    MapObject * mapObj = NULL;
+
+    while (sub_020625B0(mapObjMan, &mapObj, &v0, (1 << 0))) {
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/player_avatar.c:185]: Something like that for general map objects, haven't needed to explore it quite yet

**Review by lhearachel — COMMENTED** [review]
Third pass; I'm generally good to merge this after these changes are met.

File: src/map_object.c [hunk 19]
```diff
@@ -203,84 +201,84 @@ void sub_0206184C (MapObjectManager * mapObjMan, int param1, int param2, int par
...
 
-            if (v0 != NULL) {
-                sub_02062714(v0, param2, v2);
-                return v0;
+            if (mapObj != NULL) {
+                sub_02062714(mapObj, param2, v2);
+                return mapObj;
             }
         }
     }
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/map_object.c]: Looks like none of the weird scopes were needed, prolly were artifacts of debug code?

[the remark this answers is not in the document]
Comment (lhearachel) [reply src/map_object.c]: Probably; get rid of 'em as you come across them in functions that you document.

[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/map_object.c]: Sounds good, just cleared a few more I saw and pushed. Sorry bout this being so long I'll prolly just keep a PR perma open after this till I finish my side project so the commits don't pile up *too* much

**Review by lhearachel — APPROVED** [review]

**Pull request merged**